File: Applications/MIDIMonitor/SMMFastAnimatingWindow.m
#import "SMMFastAnimatingWindow.h"


@implementation SMMFastAnimatingWindow

- (void)awakeFromNib;
{
    if ([[self superclass] instancesRespondToSelector:@selector(awakeFromNib)])
        [super awakeFromNib];

    animationResizeTimeScaleFactor = 0.75;
}

- (NSTimeInterval)animationResizeTime:(NSRect)newFrame;
{
    return [super animationResizeTime:newFrame] * animationResizeTimeScaleFactor;
}

- (double)animationResizeTimeScaleFactor;
{
    return animationResizeTimeScaleFactor;
}

- (void)setAnimationResizeTimeScaleFactor:(double)value;
{
    animationResizeTimeScaleFactor = value;
}

@end

File: Applications/MIDIMonitor/SMMFastAnimatingWindow.swift
//
//  SMMFastAnimatingWindow.swift
//  MIDIMonitor
//
//  Created by Kurt Revis on 12/13/20.
//

import Cocoa

class SMMFastAnimatingWindow: NSWindow {

    private var animationResizeTimeScaleFactor = 0.75

    override func animationResizeTime(_ newFrame: NSRect) -> TimeInterval {
        return super.animationResizeTime(newFrame) * animationResizeTimeScaleFactor
    }

}
